src/modules/billing/controller.js

```diff
@@ -41,7 +41,8 @@ const postCreateBatch = async (request, h) => {
 
   // add a new job to the queue so that the batch can be filled
   // with charge versions
-  populateBillingBatchJob.publish(batchEvent.event_id);
+  const message = populateBillingBatchJob.createMessage(batchEvent.event_id);
+  await request.messageQueue.publish(message);
```

Comment:
I don't think we need to wait for this to resolve here.

e.g.

`request.messageQueue.publish(message);` over `await request.messageQueue.publish(message);`

Reply:
Without the await, I think it might be possible for the publish() method to throw an error but the controller still return 202

Reply:
Good point!